hc-bar/js/bar.js

```diff
@@ -0,0 +1,314 @@
+var series = [];
+
+Highcharts.data({
+  // Load Data in from Google Sheets
+  googleSpreadsheetKey: "1AKM59m3iaOSSQgVsIoDcSWxjcpX781JIuBwB9YdDgFs",
+  googleSpreadsheetWorksheet: 1,
+  switchRowsAndColumns: true,
+  parsed: function parsed(columns) {
+    // set default values
+    var drilldown = "";
+    var dataObject = { other: {}, strike: {}, intercept: {} };
+    var dataArray = [];
+    var drilldownObject = {};
+
+    columns.forEach(function (row, index) {
+      // Skip header row and title row
+      if (index == 0 || index == 1) {
+        return;
+      }
+      // Get event value for this row
+      var interceptColored = row[10].toLowerCase().indexOf("yellowgreen") > -1;
+      var strikeColored = row[10].toLowerCase().indexOf("indianred") > -1;
+      if (interceptColored) {
+        eventRow = "intercept";
+      } else if (strikeColored) {
+        eventRow = "strike";
+      } else {
+        eventRow = "other";
+      }
+      // Remove links from the text that will appear in the tooltip
+      var eventText = row[10].substr(0, row[10].indexOf("<a "));
+      // Get series date as year month 01 for this row
+      var seriesDate = new Date(row[0], row[1] - 1, "01").getTime();
+      // Get drilldown date as year month day for this row
+      var drilldownDate = new Date(row[0], row[1] - 1, row[2]).getTime();
+      // Get the drilldown value for this row
+      var drilldownRow = row[1] + "*" + row[0] + "*" + eventRow;
+      // If this event doesn't have this drilldown, create it
+      if (!dataObject[eventRow][drilldownRow]) {
+        dataObject[eventRow][drilldownRow] = {
+          y: 0,
+          drilldown: drilldownRow,
+          name: seriesDate,
+          x: seriesDate
+        };
+      }
+      // Increase value of y for every instance of the drilldown within the event object
+      dataObject[eventRow][drilldownRow].y += 1;
+
+      // populate drilldown series
+      if (!drilldownObject[drilldownRow]) {
+        drilldownObject[drilldownRow] = {
+          name: eventRow,
+          id: drilldownRow,
+          data: [],
+          yAxis: 1,
+          pointWidth: 20,
+          pointRange: 1
+        };
+      }
+      // Populate drilldown data
+      drilldownObject[drilldownRow].data.push({
+        x: drilldownDate,
+        y: 1,
+        toolHeader: row[9],
+        toolText: eventText
+      });
+    });
+    // Correct formatting
+    // Create array of event types
+    var seriesNames = Object.keys(dataObject);
+    // Create arrays of the data for each event
+    dataArray = Object.values(dataObject);
+    for (var i = 0; i < dataArray.length; i++) {
+      // Remove extra object level
+      var dataPoints = Object.values(dataArray[i]);
+      // Get series color based on event type
+      var eventColor = "";
+      if (seriesNames[i] == "intercept") {
+        eventColor = "#9acd32";
+      } else if (seriesNames[i] == "strike") {
+        eventColor = "#954950";
+      } else {
```

Comment:
On line 96 & 98 use a triple === as your are checking for an exact comparison for best practice.